packages/opentelemetry-semantic-conventions/src/resource/SemanticResourceAttributes.ts

```diff
@@ -197,411 +879,968 @@ As an alternative, consider setting `faas.id` as a span attribute instead.
+    SEMRESATTRS_K8S_REPLICASET_NAME,
+    SEMRESATTRS_K8S_DEPLOYMENT_UID,
+    SEMRESATTRS_K8S_DEPLOYMENT_NAME,
+    SEMRESATTRS_K8S_STATEFULSET_UID,
+    SEMRESATTRS_K8S_STATEFULSET_NAME,
+    SEMRESATTRS_K8S_DAEMONSET_UID,
+    SEMRESATTRS_K8S_DAEMONSET_NAME,
+    SEMRESATTRS_K8S_JOB_UID,
+    SEMRESATTRS_K8S_JOB_NAME,
+    SEMRESATTRS_K8S_CRONJOB_UID,
+    SEMRESATTRS_K8S_CRONJOB_NAME,
+    SEMRESATTRS_OS_TYPE,
+    SEMRESATTRS_OS_DESCRIPTION,
+    SEMRESATTRS_OS_NAME,
+    SEMRESATTRS_OS_VERSION,
+    SEMRESATTRS_PROCESS_PID,
+    SEMRESATTRS_PROCESS_EXECUTABLE_NAME,
+    SEMRESATTRS_PROCESS_EXECUTABLE_PATH,
+    SEMRESATTRS_PROCESS_COMMAND,
+    SEMRESATTRS_PROCESS_COMMAND_LINE,
+    SEMRESATTRS_PROCESS_COMMAND_ARGS,
+    SEMRESATTRS_PROCESS_OWNER,
+    SEMRESATTRS_PROCESS_RUNTIME_NAME,
+    SEMRESATTRS_PROCESS_RUNTIME_VERSION,
+    SEMRESATTRS_PROCESS_RUNTIME_DESCRIPTION,
+    SEMRESATTRS_SERVICE_NAME,
+    SEMRESATTRS_SERVICE_NAMESPACE,
+    SEMRESATTRS_SERVICE_INSTANCE_ID,
+    SEMRESATTRS_SERVICE_VERSION,
+    SEMRESATTRS_TELEMETRY_SDK_NAME,
+    SEMRESATTRS_TELEMETRY_SDK_LANGUAGE,
+    SEMRESATTRS_TELEMETRY_SDK_VERSION,
+    SEMRESATTRS_TELEMETRY_AUTO_VERSION,
+    SEMRESATTRS_WEBENGINE_NAME,
+    SEMRESATTRS_WEBENGINE_VERSION,
+    SEMRESATTRS_WEBENGINE_DESCRIPTION,
+  ]);
+
+/* ----------------------------------------------------------------------------------------------------------
+ * Constant values for CloudProviderValues enum definition
+ *
+ * Name of the cloud provider.
+ * ---------------------------------------------------------------------------------------------------------- */
+
+/**
+ * Name of the cloud provider.
+ */
+export const CLOUDPROVIDERVALUES_ALIBABA_CLOUD = 'alibaba_cloud';
+
+/**
+ * Name of the cloud provider.
+ */
+export const CLOUDPROVIDERVALUES_AWS = AWS as 'aws';
+
+/**
+ * Name of the cloud provider.
+ */
+export const CLOUDPROVIDERVALUES_AZURE = AZURE as 'azure';
+
+/**
+ * Name of the cloud provider.
+ */
+export const CLOUDPROVIDERVALUES_GCP = 'gcp';
+
+/**
+ * Identifies the Values for CloudProviderValues enum definition
+ *
+ * Name of the cloud provider.
+ * @deprecated Use the CLOUDPROVIDERVALUES_XXXXX constants rather than the CloudProviderValues.XXXXX for bundle minification.
+ */
+export type CloudProviderValues = {
   /** Alibaba Cloud. */
-  ALIBABA_CLOUD: 'alibaba_cloud',
+  ALIBABA_CLOUD: 'alibaba_cloud';
   /** Amazon Web Services. */
-  AWS: 'aws',
+  AWS: 'aws';
   /** Microsoft Azure. */
-  AZURE: 'azure',
+  AZURE: 'azure';
   /** Google Cloud Platform. */
-  GCP: 'gcp',
-} as const;
-export type CloudProviderValues =
-  (typeof CloudProviderValues)[keyof typeof CloudProviderValues];
+  GCP: 'gcp';
+};
+
+/**
+ * The constant map of values for CloudProviderValues.
+ * @deprecated Use the CLOUDPROVIDERVALUES_XXXXX constants rather than the CloudProviderValues.XXXXX for bundle minification.
+ */
+export const CloudProviderValues: CloudProviderValues = {
```

Comment:
Not using the same `createConstMap` as above because the `key` cannot not always be derived from the value of the enum types 😢 , so this still needs to be defined with the unminifiable `key`, but it can still references the new exported variable -- so the value is only included once (per type).

Reply:
Thats a good idea, as it will help reduce the size further. I'll attempt to do this again.
From memory there was "something" that caused my original attempt to fail, possible due to either the different types or the names not matching the values (I don't remember).

Reply:
Latest version updates and defines enum values using the `createConstMap` if possible (the Http values are defined differently where the "name" is `HTTP_1_0` but the value is `1.0`